packages/prelude/src/array.ts

```diff
@@ -2,9 +2,10 @@
 
 // Non-mutating array operations
 
-// cons :: a -> [a] -> [a]
-// a with x prepended
-export function cons (x, a) {
+/**
+ * a with x prepended
+ */
+export function cons <A> (x: A, a: A[]): A[] {
```

Comment:
I think it'd be especially nice to use readonly arrays in prelude wherever possible, since one of the goals of prelude is to provide fast, immutable array operations.

Reply:
I've seen a separate #156 tracking that. Should we merge it into migration to TS?

Reply:
That'd be great.  Thank you.

Reply:
Done. I pushed it as a `feat:` though. Note that I also added `concat` to `prelude`.